doc/dev/release.md

```diff
@@ -184,8 +184,6 @@ Commit changes to the following six files:
 
 Create a new PR for `release-v1.3.0`.
 
-**Note:** CI will not pass for this commit because the new version `v1.3.0` does not exist yet in GitHub.
```

Comment:
Did we figure out how to pass CI for release branches referencing tags that don't exist?

Reply:
No, but once you push the tag the build restarts for that commit. And this comment was actually talking about the PR's test failing, not the branch, which doesn't happen anymore.